rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize_crop_mirror.cpp

```diff
@@ -27,60 +27,84 @@ THE SOFTWARE.
+    _mirror.create_array(_graph, VX_TYPE_UINT32, _batch_size);
+    create_crop_tensor(_crop_tensor, &_crop_coordinates);
+    vx_scalar interpolation_vx = vxCreateScalar(vxGetContext((vx_reference)_graph->get()), VX_TYPE_INT32, &_interpolation_type);
+    _node = vxExtRppResizeCropMirror(_graph->get(), _inputs[0]->handle(), _crop_tensor, _outputs[0]->handle(), _dst_roi_width, 
+                                     _dst_roi_height, _mirror.default_array(), interpolation_vx, _input_layout, _output_layout, _roi_type);
 
     vx_status status;
     if((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
-        THROW("Error adding the resize crop mirror resize node (vxExtrppNode_ResizeCropbatchPD    ) failed: "+TOSTR(status))
+        THROW("Error adding the resize crop mirror node (vxExtRppResizeCropMirror) failed: " + TOSTR(status))
 }
 
-void ResizeCropMirrorNode::update_node()
-{
+void ResizeCropMirrorNode::update_node() {
     _crop_param->set_image_dimensions(_inputs[0]->info().get_roi());
     _crop_param->update_array();
+    std::vector<uint32_t> crop_h_dims, crop_w_dims;
+    _crop_param->get_crop_dimensions(crop_w_dims, crop_h_dims);
+    _outputs[0]->update_tensor_roi(crop_w_dims, crop_h_dims);
+    _mirror.update_array();
+    
+    // Obtain the crop coordinates and update the roi
+    auto x1 = _crop_param->get_x1_arr_val();
+    auto y1 = _crop_param->get_y1_arr_val();
+    RocalROI *crop_dims = static_cast<RocalROI *>(_crop_coordinates);
+    for(unsigned i = 0; i < _batch_size; i++) {
+        crop_dims[i].x1 = x1[i];
+        crop_dims[i].y1 = y1[i];
+        crop_dims[i].x2 = crop_w_dims[i];
+        crop_dims[i].y2 = crop_h_dims[i];
+    }
 }
 
-void ResizeCropMirrorNode::init(unsigned int crop_h, unsigned int crop_w, IntParam *mirror)
+void ResizeCropMirrorNode::init(unsigned int crop_h, unsigned int crop_w, IntParam *mirror, RocalResizeInterpolationType interpolation_type)
 {
     _crop_param->crop_w = crop_w;
     _crop_param->crop_h = crop_h;
-    _crop_param->x1     = 0;
-    _crop_param->y1     = 0;
+    _crop_param->x1 = 0;
+    _crop_param->y1 = 0;
     _mirror.set_param(core(mirror));
+    _interpolation_type = static_cast<int>(interpolation_type);
 }
 
-
-void ResizeCropMirrorNode::init(FloatParam *crop_h_factor, FloatParam  *crop_w_factor, IntParam *mirror)
-{
+void ResizeCropMirrorNode::init(FloatParam *crop_h_factor, FloatParam *crop_w_factor, IntParam *mirror, RocalResizeInterpolationType interpolation_type) {
     _crop_param->set_crop_height_factor(core(crop_h_factor));
     _crop_param->set_crop_width_factor(core(crop_w_factor));
     _crop_param->set_random();
     _mirror.set_param(core(mirror));
+    _interpolation_type = static_cast<int>(interpolation_type);
 }
 
+ResizeCropMirrorNode::~ResizeCropMirrorNode() {
+    if (_inputs[0]->info().mem_type() == RocalMemType::HIP) {
+#if ENABLE_HIP
```

Comment:
this is not how we need to check if the pipeline is in cpu or device. For example ENABLE_HIP is default mode and user can create a pipeline on cpu or gpu. Need to use the device parameter of pipeline instead

Reply:
We are making use of the mem_type() that is set to the tensor. If it is a HIP tensor then HIP mem is allocated and also removed for the ROI